vllm/entrypoints/openai/serving_chat.py

```diff
@@ -158,6 +158,21 @@ async def create_chat_completion(
             logger.error("Error in loading multi-modal data: %s", e)
             return self.create_error_response(str(e))
 
+        # validation for OpenAI tools
+        try:
+            # tool_choice = "required" is not supported
+            assert(request.tool_choice != 'required', 'tool_choice="required" is not supported.')
```

Comment:
Hmm - I wasn't aware of the issue with `assert` - I used this because I think I saw it somewhere else, so i will change that too. I'm pretty much a hobbyist as far as python goes, so I'm not really familiar with internals like this (I'm a Node/TS guy most of the time). 

I'll patch this :)